src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(MakeLocalFunctionStaticCodeRefactoringProvider)), Shared]
+    internal sealed class MakeLocalFunctionStaticCodeRefactoringProvider : CodeRefactoringProvider
+    {
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var (document, textSpan, cancellationToken) = context;
+
+            var localFunction = await context.TryGetRelevantNodeAsync<LocalFunctionStatementSyntax>().ConfigureAwait(false);
+            if (localFunction == null)
+            {
+                return;
+            }
+
+            if (!MakeLocalFunctionStaticHelper.IsStaticLocalFunctionSupported(localFunction.SyntaxTree))
+            {
+                return;
+            }
+
+            if (localFunction.Modifiers.Any(SyntaxKind.StaticKeyword))
+            {
+                return;
+            }
+
+            var semanticModel = (await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false))!;
+
+            if (MakeLocalFunctionStaticHelper.TryGetCaputuredSymbols(localFunction, semanticModel, out var captures) &&
+                captures.Length > 0 &&
+                !captures.Any(s => s.IsThisParameter()))
```

Comment:
Note that this currently doesn't support local functions that captures the implicit `this` parameter, i.e. referencing fields, properties, etc. Same for the codefix in this PR.

Reply:
i feel like that wouldn't be hard to support fwiw.

Reply:
Will implement this in a follow up PR